filters.js

```diff
@@ -447,6 +474,31 @@ function operationOnDates(v, arg, operation) {
   }
 }
 
+function toCurrency(currValue, currType) {
+  if (
+    isValidNumber(currValue) &&
+    _.isString(currType)
+  ) {
+    return { value: currValue, type: currType };
+  }
+  throw new Error("invalid currency value or type");
+}
+
+function toDuration(durValue, durType) {
+  if (
+    isValidNumber(durValue) &&
+    _.isString(durType)
+  ) {
+    const durationType = durType.toUpperCase();
+
```

Comment:
We should add a new if statement which checks whether the passed `durType` is allowed or not.

Reply:
@sanket0896 modified the code, please see below

```
function toDuration(durValue, durType) {
  if (
    isValidNumber(durValue) &&
    _.isString(durType)
  ) {
    const durationType = durType.toUpperCase();

    if(Object.values(DURATION_TYPES).includes(durationType)){
      return {
        value: durValue,
        type: durationType,
        days: calculateDaysFromDurValueAndType(durValue, durationType)
      };
    }
  }
  throw new Error("invalid duration value or type");
}
```

this does not log error as per `calculateDaysFromDurValueAndType()` which is

```
throw new Error(
        `duration type of ${durType} found to be incorrect` +
          `while calculating days from durValue and durType`
      );
```

hence test fails as per currently written code (for duration type fortnight).

I am assuming I should modify the test case to throw `invalid duration value or type`, but still keep the error in default of `calculateDaysFromDurValueAndType()`?

cc @anirudhvarma12

Reply:
yes, that's correct. Throw an error after the inner if as well